nifi-nar-bundles/nifi-standard-bundle/nifi-transform-json-ui/src/main/java/org/apache/nifi/web/standard/api/processor/ProcessorResource.java

```diff
@@ -0,0 +1,100 @@
+        final NiFiWebConfigurationContext nifiWebContext = getWebConfigurationContext();
+        final ComponentDetails componentDetails = CustomUIUtils.getComponentDetails(nifiWebContext, processorId, request);
+        final Response.ResponseBuilder response = CustomUIUtils.applyCacheControl(Response.ok(componentDetails.getProperties()));
+        return response.build();
+    }
+
+    @GET
+    @Produces({MediaType.APPLICATION_JSON})
+    @Path("/type")
+    public Response getType(@QueryParam("processorId") final String processorId) {
+        final NiFiWebConfigurationContext nifiWebContext = getWebConfigurationContext();
+        final ComponentDetails componentDetails = CustomUIUtils.getComponentDetails(nifiWebContext, processorId, request);
+        final Response.ResponseBuilder response = CustomUIUtils.applyCacheControl(Response.ok(componentDetails.getType()));
+        return response.build();
+    }
+
+    @GET
+    @Produces({MediaType.APPLICATION_JSON})
+    @Path("/descriptors")
+    public Response getDescriptors(@QueryParam("processorId") final String processorId) {
+        final NiFiWebConfigurationContext nifiWebContext = getWebConfigurationContext();
+        final ComponentDetails componentDetails = CustomUIUtils.getComponentDetails(nifiWebContext, processorId, request);
+        final Map<String,ComponentDescriptor> descriptors = componentDetails.getDescriptors();
+        final Response.ResponseBuilder response = CustomUIUtils.applyCacheControl(Response.ok(descriptors));
+        return response.build();
+    }
+
+    @GET
+    @Produces({MediaType.APPLICATION_JSON})
+    @Path("/details")
+    public Response getDetails(@QueryParam("processorId") final String processorId) {
+        final NiFiWebConfigurationContext nifiWebContext = getWebConfigurationContext();
+        final ComponentDetails componentDetails = CustomUIUtils.getComponentDetails(nifiWebContext, processorId, request);
+        final Response.ResponseBuilder response = CustomUIUtils.applyCacheControl(Response.ok(componentDetails));
+        return response.build();
+    }
+
+    @POST
+    @Produces({MediaType.APPLICATION_JSON})
+    @Consumes({MediaType.APPLICATION_JSON})
+    @Path("/properties")
+    public Response setProperties(@QueryParam("processorId") final String processorId, @QueryParam("revisionId") final Long revisionId,
+                                  @QueryParam("clientId") final String clientId, Map<String,String> properties){
+        final NiFiWebConfigurationContext nifiWebContext = getWebConfigurationContext();
+        final NiFiWebConfigurationRequestContext niFiRequestContext = CustomUIUtils.getRequestContext(processorId,revisionId,clientId,request);
+        final ComponentDetails componentDetails = nifiWebContext.setProperties(niFiRequestContext,properties);
+        final Response.ResponseBuilder response = CustomUIUtils.applyCacheControl(Response.ok(componentDetails));
```

Comment:
Depending on the scheme, this should return the appropriate response. Created or Ok.